src/scene/materials/standard-material.js

```diff
@@ -868,6 +884,12 @@ class StandardMaterial extends Material {
         library.register('standard', standard);
         const shader = library.getProgram('standard', options, processingOptions, this.userId);
 
+        if (this.userAttributes) {
```

Comment:
I don't think we want to do this here this way, as the getProgram internally does shader caching, and the cache key needs to know about those additional attributes, otherwise it will provide the same shader regardless of the need of the additional attributes.

I think here we need to convert userAttributes to options before calling the getProgram. That internally calls standard.js generator, where the options would be to generate the key and also generate the matching shader.

Reply:
I moved it to options builder. Would that be fine? Or do you want it to be added to options before that?